### PR TITLE
Add self tests for rsa encryption

### PR DESCRIPTION
**Pull request by slontis**
SP800-56br2 requires seperate KAT's (fips self tests) to be tested for both encryption and decryption
using the RSA primitive (i.e. no padding). This is specified in FIPS140-2 IG D.9

A copy of the methods EVP_PKEY_encrypt_init(), EVP_PKEY_encrypt(), EVP_PKEY_decrypt_init(), EVP_PKEY_decrypt()
are now in the fips module.

Removed the #ifdef FIPS_MODULE in evp_pkey_ctx_free_old_ops().

Added corruption test

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: providers/fips/self_test_data.inc [hunk 1]
```diff
@@ -955,6 +967,25 @@ static const ST_KAT_PARAM rsa_key[] = {
...
 };
 
+static const ST_KAT_PARAM rsa_pub_key[] = {
+    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_N, rsa_n),
+    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_E, rsa_e),
+    ST_KAT_PARAM_END()
+};
+
+static const ST_KAT_PARAM rsa_priv_key[] = {
+    ST_KAT_PARAM_BIGNUM(OSSL_PKEY_PARAM_RSA_N, rsa_n),
```
Comment (t8m) [inline providers/fips/self_test_data.inc:977]: No CRT parameters?

Comment (slontis) [reply providers/fips/self_test_data.inc:977]: They are not used by a primitive test (as far as I could see) - this case is with no padding,, So it is just m^e mod n

Comment (t8m) [reply providers/fips/self_test_data.inc:977]: Really? I believe in the end for the private key decryption the rsa_ossl_private_decrypt() should be called and that will do the CRT computation if the CRT parameters are present.

So there probably should be actually two tests done for the private key operation - one with CRT parameters and one without.

Comment (slontis) [reply providers/fips/self_test_data.inc:977]: You are right I was looking at the public one..

Comment (slontis) [reply providers/fips/self_test_data.inc:977]: Changed test to use a crt key also for private decryption. Removed the extra encrypt/decrypt data. I think it makes more sense testing the same data that was used for encryption for the decryption test.

**Comment by slontis** [comment]
travis error is just a timeout.

**Comment by paulidale** [comment]
Ping for @openssl/committers review

File: providers/fips/self_test_kats.c [hunk 2]
```diff
@@ -504,6 +502,93 @@ static int self_test_sign(const ST_KAT_SIGN *t,
     return ret;
 }
 
+/*
+ * Test a encrypt or decrypt KAT..
```
Comment (t8m) [inline providers/fips/self_test_kats.c:506]: Nit: an encrypt.

File: test/recipes/03-test_fipsinstall.t [hunk 3]
```diff
@@ -225,6 +225,16 @@ SKIP: {
        "fipsinstall fails when the signature result is corrupted");
 }
 
+# corrupt a Asymmetric cipher test
```
Comment (t8m) [inline test/recipes/03-test_fipsinstall.t:228]: Nit: an

**Review by t8m — APPROVED** [review]
LGTM, the two comment nits can be fixed without reapproval.

**Comment by slontis** [comment]
Trivial NITS updated.

**Comment by slontis** [comment]
Thanks.. Merged to master.

**Pull request closed without merging**